crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_numeric_union.rs

```diff
@@ -47,16 +61,21 @@ pub struct RedundantNumericUnion {
     redundancy: Redundancy,
 }
 
-impl Violation for RedundantNumericUnion {
+impl AlwaysFixableViolation for RedundantNumericUnion {
     #[derive_message_formats]
     fn message(&self) -> String {
         let (subtype, supertype) = match self.redundancy {
+            Redundancy::IntFloatComplex => ("int | float", "complex"),
```

Comment:
Emitting multiple messages for the same union, such as "Use `complex` instead of `int | complex`" and  "Use `complex` instead of `float | complex`" gets confusing when there is a fix presented:

```diff
-    def bad4(self, arg: Union[float | complex, int]) -> None: ...  # PYI041
+    def bad4(self, arg: complex) -> None: ...  # PYI041
```

With `IntFloatComplex` we can assign an unique violation for each combination.